internal/proto/controller/api/resources/credentiallibraries/v1/credential_library.proto

```diff
@@ -7,60 +7,74 @@ option go_package = "github.com/hashicorp/boundary/sdk/pbs/controller/api/resour
 import "controller/custom_options/v1/options.proto";
 
 // CredentialLibrary contains all fields related to an Credential Library resource
 message CredentialLibrary {
   // Output only. The ID of the Credential Library.
-  string id = 10;
+  string id = 10; // @gotags: `class:"public"`
 
   // The ID of the Credential Store of which this Credential Library is a part.
-  string credential_store_id = 20 [json_name = "credential_store_id"];
+  string credential_store_id = 20 [
+    json_name = "credential_store_id",
+    (controller.custom_options.v1.subtype_source_id) = true
+  ]; // @gotags: `class:"public"`
 
   // Output only. Scope information for this Credential Library.
-  resources.scopes.v1.ScopeInfo scope = 30;
+  resources.scopes.v1.ScopeInfo scope = 30; // @gotags: `class:"public"`
 
   // Optional name for identification purposes.
-  google.protobuf.StringValue name = 40 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "name" that: "Name" }];
+  google.protobuf.StringValue name = 40 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "name" that: "Name" }]; // @gotags: `class:"public"`
 
   // Optional user-set description for identification purposes.
-  google.protobuf.StringValue description = 50 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "description" that: "Description" }];
+  google.protobuf.StringValue description = 50 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "description" that: "Description" }]; // @gotags: `class:"public"`
 
   // Output only. The time this resource was created.
-  google.protobuf.Timestamp created_time = 60 [json_name = "created_time"];
+  google.protobuf.Timestamp created_time = 60 [json_name = "created_time"]; // @gotags: `class:"public"`
 
   // Output only. The time this resource was last updated.
-  google.protobuf.Timestamp updated_time = 70 [json_name = "updated_time"];
+  google.protobuf.Timestamp updated_time = 70 [json_name = "updated_time"]; // @gotags: `class:"public"`
 
   // Version is used in mutation requests, after the initial creation, to ensure this resource has not changed.
   // The mutation will fail if the version does not match the latest known good version.
-  uint32 version = 80;
+  uint32 version = 80; // @gotags: `class:"public"`
 
   // The Credential Library type.
-  string type = 90;
-
-  // The attributes that are applicable for the specific Credential Library type.
-  google.protobuf.Struct attributes = 100 [(custom_options.v1.generate_sdk_option) = true];
+  string type = 90; // @gotags: `class:"public"`
+
+  oneof attrs {
+    // The attributes that are applicable for the specific Credential Library type.
+    google.protobuf.Struct attributes = 100 [
+      (custom_options.v1.generate_sdk_option) = true,
+      (custom_options.v1.subtype) = "default"
+    ];
```

Comment:
We can classify this field, I think? It is a well known type, at least.

Reply:
iirc it is treated as a map. so the classification would not have an effect.

Reply:
Ah, seems reasonable